### PR TITLE
Add accordion functionality to 'Whats on' filters

### PR DESCRIPTION
**Pull request by janehughes**
Ticket URL: n/a (yet)

## About these changes

Adds accordion/dropdown for the filters on Whats On landing page
Progressive enhancement - ensures the relevant classes hide the accordion content only once JS has loaded

## How to check these changes

Where possible, provide guidance to help your reviewer

## Before assigning to reviewer, please make sure you have

- [ ] Checked things thoroughly before handing over to reviewer
- [ ] Checked PR title starts with ticket number as per project conventions to help us keep track of changes
- [ ] Ensured that PR includes only commits relevant to the ticket
- [ ] Waited for all CI jobs to pass before requesting a review
- [ ] Added/updated tests and documentation where relevant

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: sass/includes/_accordion.scss [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        text-decoration: none;
+        width: 100%;
+
+        &:hover {
+            cursor: pointer;
+        }
+
+        /* stylelint-disable */
+        &:focus {
+            outline: 0; // focus style implemented directly on text instead
```
Comment (helenb) [inline sass/includes/_accordion.scss:39]: I don't see any focus state when the accordions are closed

File: sass/includes/_filters.scss [hunk 2]
```diff
@@ -20,7 +22,8 @@
 
     &__fieldset {
         flex-shrink: 0;
-        border: 1px solid $color__black;
+        background-color: $color__white;
```
Comment (helenb) [inline sass/includes/_filters.scss:25]: Query: is the border still visible if there is no javascript used on the page?

File: scripts/src/modules/accordion.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                this.button.setAttribute("aria-expanded", "false");
+                this.content.setAttribute("aria-hidden", "true");
+                this.accordion.classList.remove("is-open");
+                this.accordion.classList.add("is-closed");
+                open = false;
+            } else {
+                this.button.setAttribute("aria-expanded", "true");
+                this.content.setAttribute("aria-hidden", "false");
+                this.accordion.classList.add("is-open");
+                this.accordion.classList.removed("is-closed");
```
Comment (helenb) [inline scripts/src/modules/accordion.js:29]: As discussed this should be `classList.remove`

File: sass/includes/_filters.scss (hunk 2 above)
Comment (helenb) [reply sass/includes/_filters.scss:25]: I've tested without js now, and it looks like we still have the white background. I think this is fine - Nick wanted the black outline but if that is tricky to implement lets stick with what we have.

Comment (janehughes) [reply sass/includes/_filters.scss:25]: The black border isn't in the figma designs so I removed it - I can have another look at this though if it's required, @nicklee please let me know!

File: sass/includes/_accordion.scss (hunk 1 above)
Comment (janehughes) [reply sass/includes/_accordion.scss]: Resolved, thank you for flagging!

File: scripts/src/modules/accordion.js (hunk 3 above)
Comment (janehughes) [reply scripts/src/modules/accordion.js]: 👍  thanks!

**Review by helenb — APPROVED** [review]
Looks good to me now if Nick is happy with the styling change

**Pull request merged**